db/funds.go

```diff
@@ -45,7 +45,7 @@ func (f *FundsDB) Untag(ctx context.Context, dealUuid uuid.UUID) (abi.TokenAmoun
 	err := row.Scan(&collat.marshalled, &pubMsg.marshalled)
 	if err != nil {
 		if err == sql.ErrNoRows {
-			return abi.NewTokenAmount(0), nil
+			return abi.NewTokenAmount(0), ErrDealNotFound
```

Comment:
I don't think we need to return an error if the deal has already been untagged. It should make the calling code simpler if we just return nil.

Reply:
@dirkmc Trying to untag a deal that does NOT exist means something has gone wrong somewhere. If the caller does NOT care about it, it is the calling code's responsibility to handle it. However, this layer should NOT ignore that error and I think most callers should log it so we can debug what went wrong.